packages/firestore/test/unit/local/index_manager.test.ts

```diff
@@ -886,6 +886,23 @@ describe('IndexedDbIndexManager', async () => {
     await verifyResults(queryWithRestrictedBound, 'coll/val4');
   });
 
+  it('cannot expand result set from a cursor', async () => {
+    await indexManager.addFieldIndex(
+      fieldIndex('coll', { fields: [['c', IndexKind.ASCENDING]] })
+    );
+    await addDoc('coll/val1', { 'a': 1, 'b': 1, 'c': 3 });
+    await addDoc('coll/val2', { 'a': 2, 'b': 2, 'c': 2 });
+
+    const testingQuery = queryWithStartAt(
+      queryWithAddedOrderBy(
+        queryWithAddedFilter(query('coll'), filter('c', '>', 2)),
+        orderBy('c', 'asc')
+      ),
+      bound([2], /* inclusive= */ true)
+    );
+    await verifyResults(testingQuery, 'coll/val1');
+  });
+
```

Comment:
can you also add another test (or just add to the existing test) for the other direction?  e.g. `c < 2 order by c DESC` with bound 2 inclusive.

Reply:
And that uncovers one more bug.